### PR TITLE
Remove implicit switch-ons of "compute.ops_on_diff_frames"

### PR DESCRIPTION
**Pull request by xinrong-databricks**
## Proposal
- Remove implicit switch-ons of "compute.ops_on_diff_frames".
- Fix doctests
- Fix unit tests

### CODE REVIEW

**Comment by xinrong-databricks** [comment]
Shall we modify the docstrings of each touched function to reflect this change?

**Comment by HyukjinKwon** [comment]
I tend to agree with this change ... but to confirm WDYT @ueshin and @itholic?

**Comment by itholic** [comment]
I agree, too!

> Shall we modify the docstrings of each touched function to reflect this change?

For this, IMO, I think we don't necessarily need to modify the docstrings since it already shows a proper error message - `Cannot combine the series or dataframe because it comes from a different dataframe. In order to allow this operation, enable 'compute.ops_on_diff_frames' option.` -.

**Review by ueshin — APPROVED** [review]
LGTM.

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -5590,11 +5593,12 @@ def compare(
...
+
+        >>> reset_option("compute.ops_on_diff_frames")
         """
-        with option_context("compute.ops_on_diff_frames", True):
-            if not self.index.equals(other.index):
-                raise ValueError("Can only compare identically-labeled Series objects")
-            combined = combine_frames(self.to_frame(), other.to_frame())
+        if not self.index.equals(other.index):
+            raise ValueError("Can only compare identically-labeled Series objects")
+        combined = combine_frames(self.to_frame(), other.to_frame())
```
Comment (ueshin) [inline databricks/koalas/series.py:5601]: Btw, we should consider other is from the same anchor:

```py
>>> s1 = ks.Series(["a", "b", "c", "d", "e"])
>>> s1.compare(s1)
Traceback (most recent call last):
...
AssertionError: We don't need to combine. `this` and `that` are same.
```

Comment (xinrong-databricks) [reply databricks/koalas/series.py:5601]: Good idea! Let me file a new PR for this.

**Pull request merged**